### PR TITLE
SendLocation for Mobile Nav

### PR DESCRIPTION
**Pull request by robinmk**
This proposal is about extending the capability for apps to receive and service SendLocation requests from other apps which is currently limited to the Head Unit's embedded navigation system.

### CODE REVIEW

**Comment by robinmk** [comment]
@theresalech , I've some assets files which needs to be referenced by this proposal. Where do I add those?

**Comment by joeljfischer** [comment]
@robinmk Check this: https://github.com/smartdevicelink/sdl_evolution/tree/master/assets/proposals

**Comment by robinmk** [comment]
@joeljfischer, I've added the assets via another PR. Can you please check: https://github.com/smartdevicelink/sdl_evolution/pull/369

**Comment by joeljfischer** [comment]
@robinmk They should all be a part of one PR, e.g. https://github.com/smartdevicelink/sdl_evolution/pull/245

Also I'm not sure what abc.txt is meant to do?

**Comment by robinmk** [comment]
@joeljfischer , Thank you.
I have added the assets as part of this PR and deleted the abc.txt file.
I will be closing the second PR that I created for the assets.

**Comment by robinmk** [comment]
@theresalech , Please mark as ready for review.

File: proposals/NNNN-SendLocation-for-Mobile-Nav.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+ The concept introduced here is the ability for the system (HU) to send a request to the app and for the app to provide the response. Today, we have ability for the app to send a request and for the system to respond and the ability for the system to send a notification (i.e. no response expected) to the application.
+
+## Impact on existing code
+### Design Details
+
+1) *There must be a mechanism for the system to notify apps that it has an enhanced capability (by virtue of another app providing this capability)*  
+Today, applications can query the system for its supported capability through the `GetSystemCapability` request. Now, with this proposal, applications ("Service applications")would have the ability to enhance the system's capability by providing certain services. To other apps, this should appear as a capability of the system itself and the apps should request for services to the system as they normally would. Hence it becomes necessary for the system to notify all registered applications when its capability has been enhanced so that the applications can take advantage of these capabilities.
```
Comment (theresalech) [inline proposals/NNNN-SendLocation-for-Mobile-Nav.md]: Please add a space after "(Service applications")" and before "would"

File: proposals/NNNN-SendLocation-for-Mobile-Nav.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  ```xml
+  <function name="OnSystemCapabilityChange" functionID="OnSystemCapabilityChangeID" messagetype="notification">
+     <description>Notification from the module when the module's capability has changed.</description>
+     <param name="systemCapability" type="SystemCapability" mandatory="false" array="true">
+     </param>
+ </function>
+  ```
+  i) SDL Core shall send this notification to all registered applications.  
+  ii) SDL Core shall send this notification only when the Service Application becomes active. In the case of SendLocation for Mobile Nav, the app can service the request only when the Service Application (Mobile Nav) app is active (i.e. in HMI state of FULL/LIMITED).  
+  iii) SDL Core shall only send this notificaiton when there is a change in system's capability. For example, if the system has an embedded navigation source which supports SendLocation and then the user activates a Mobile Nav app which also supports SendLocation then there is no actual change in the capability of the system and hence SDL Core shall not send the notification in this case.
```
Comment (theresalech) [inline proposals/NNNN-SendLocation-for-Mobile-Nav.md]: Please fix spelling for "notification"

File: proposals/NNNN-SendLocation-for-Mobile-Nav.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+c)When SDL Core re-routes the request to the Service Application, it should add a valid Correlation ID unlike the '-1' value it sends for notifications. The suggestion is for SDL Core to maintain its own Correlation ID which will be incremented by 1 whenever SDL Core will need to send a request out to a Service Application. The Service Application should provide the same Correlation ID in its response.  
+d)SDL Core should start a timer waiting for the response from the Service Application. If the timer expires before the SDL Core receives a response from the Service Application, then SDL Core should provide a response of `TIMED_OUT` to the requesting application.  
+e)The value of this timer should be carefully selected to cover various scenarios.  
+f)If the Service Application is not in FULL (i.e. being actively viewed on HMI) and this application receives a SendLocation request, the Service Application can draw attention to the user by using an Alert RPC informing that there is a destination informaiton ready to be viewed on the app.
```
Comment (theresalech) [inline proposals/NNNN-SendLocation-for-Mobile-Nav.md]: Please update to "...informing that there is destination information ready to be viewed on the app."

File: proposals/NNNN-SendLocation-for-Mobile-Nav.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
+# SendLocation for Mobile Nav
+
+* Proposal: [NNNN](NNNN-SendLocation-for-Mobile-Nav.md)
```
Comment (theresalech) [inline proposals/NNNN-SendLocation-for-Mobile-Nav.md]: Please update to "[SDL-NNNN]"

**Comment by robinmk** [comment]
@theresalech , please check

**Pull request merged**